gulp/webpack.config.js

```diff
@@ -1,127 +1,103 @@
+    assert: "window.assert",
+    assertAlways: "window.assert",
+    abstract:
+        "window.assert(false, 'abstract method called of: ' + " +
+        "(this.name || (this.constructor && this.constructor.name)));",
+    G_IS_DEV: "true",
+    G_APP_ENVIRONMENT: JSON.stringify("development"),
+    G_BUILD_TIME: new Date().getTime().toString(),
+    G_BUILD_COMMIT_HASH: JSON.stringify(getRevision()),
+    G_BUILD_VERSION: JSON.stringify(getVersion()),
+    G_ALL_UI_IMAGES: JSON.stringify(getAllResourceImages()),
 
-module.exports = ({
-    watch = false,
-    standalone = false,
-    chineseVersion = false,
-    wegameVersion = false,
-    steamDemo = false,
-    gogVersion = false,
-}) => {
-    return {
-        mode: "development",
-        devtool: "cheap-source-map",
-        entry: {
-            "bundle.js": [path.resolve(__dirname, "../src/js/main.js")],
-        },
-        watch,
-        node: {
-            fs: "empty",
-        },
-        resolve: {
-            alias: {
-                "global-compression": path.resolve(__dirname, "..", "src", "js", "core", "lzstring.js"),
-            },
-        },
-        context: path.resolve(__dirname, ".."),
-        plugins: [
-            new webpack.DefinePlugin({
-                assert: "window.assert",
-                assertAlways: "window.assert",
-                abstract:
-                    "window.assert(false, 'abstract method called of: ' + (this.name || (this.constructor && this.constructor.name)));",
-                G_HAVE_ASSERT: "true",
-                G_APP_ENVIRONMENT: JSON.stringify("dev"),
-                G_CHINA_VERSION: JSON.stringify(chineseVersion),
-                G_WEGAME_VERSION: JSON.stringify(wegameVersion),
-                G_GOG_VERSION: JSON.stringify(gogVersion),
-                G_IS_DEV: "true",
-                G_IS_RELEASE: "false",
-                G_IS_BROWSER: "true",
-                G_IS_STANDALONE: JSON.stringify(standalone),
-                G_IS_STEAM_DEMO: JSON.stringify(steamDemo),
-                G_BUILD_TIME: "" + new Date().getTime(),
-                G_BUILD_COMMIT_HASH: JSON.stringify(getRevision()),
-                G_BUILD_VERSION: JSON.stringify(getVersion()),
-                G_ALL_UI_IMAGES: JSON.stringify(getAllResourceImages()),
-            }),
-
-            new CircularDependencyPlugin({
-                // exclude detection of files based on a RegExp
-                exclude: /node_modules/,
-
-                // add errors to webpack instead of warnings
-                failOnError: true,
-
-                // allow import cycles that include an asyncronous import,
-                // e.g. via import(/* webpackMode: "weak" */ './file.js')
-                allowAsyncCycles: false,
+    G_CHINA_VERSION: "false",
+    G_WEGAME_VERSION: "false",
+    G_GOG_VERSION: "false",
+    G_IS_RELEASE: "false",
+    G_IS_STANDALONE: "true",
+    G_IS_STEAM_DEMO: "false",
+    G_IS_BROWSER: "false",
+    G_HAVE_ASSERT: "true",
+};
```

Comment:
Can the CHINA, WEGAME, GOG, STEAM_DEMO globals be removed?

Reply:
Most of these code changes dealing with `/gulp` or `/src/js` are pulls from other branches, which I would rather pull from each branch directly than fix on this copy. This seems to be from #3, and I would recommend adding a comment there. If @DJ1TJOO makes any changes, ill pull them into this branch to sync the fixes

Reply:
We could do that, but you would still need to reconcile it with the changes made by Emerald to the webpack configs and check all of the ts files for these globals.  I would recommend just doing it yourself, or leaving it for another time.